### PR TITLE
[BUG] Remove enum imports from match statements

### PR DESCRIPTION
**Pull request by raunakab**
# Overview
Importing enums is extremely bad style in Rust. This can lead to nasty errors in the future when enum definitions change.

Please stop doing this.

### CODE REVIEW

**Comment by andrewgazelka** [comment]
hooray I agree 🕺

**Comment by raunakab** [comment]
> hooray I agree 🕺

@andrewgazelka This is a large problem in our codebase. I'm going to start merging in small, piecemeal PRs to address this issue.

File: src/daft-core/src/datatypes/matching.rs [hunk 1]
```diff
@@ -1,293 +1,362 @@
...
+            DataType::Time(_) => __with_ty__! { TimeType },
+            DataType::Timestamp(_, _) => __with_ty__! { TimestampType },
+            DataType::UInt16 => __with_ty__! { UInt16Type },
+            DataType::UInt32 => __with_ty__! { UInt32Type },
+            DataType::UInt64 => __with_ty__! { UInt64Type },
+            DataType::UInt8 => __with_ty__! { UInt8Type },
+            DataType::Unknown => unimplemented!("array for unknown datatype not implemented"),
+            DataType::Utf8 => __with_ty__! { Utf8Type },
+            #[cfg(feature = "python")]
+            DataType::Python => __with_ty__! { PythonType },
```
Comment (andrewgazelka) [inline src/daft-core/src/datatypes/matching.rs:49]: if you want to be even more idiomatic can do $crate::DataType::{x}

similar with these (assuming same crate)

I'd recommend this

**Review by andrewgazelka — CHANGES_REQUESTED** [review]

**Comment by andrewgazelka** [comment]
> > hooray I agree 🕺
> 
> @andrewgazelka This is a large problem in our codebase. I'm going to start merging in small, piecemeal PRs to address this issue.

Unsure if we should prio this though. 

_BigDaft_

ideally if we want to enforce something should also add to clippy lints
https://rust-lang.github.io/rust-clippy/master/index.html

File: src/daft-core/src/datatypes/matching.rs (hunk 1 above)
Comment (raunakab) [reply src/daft-core/src/datatypes/matching.rs:49]: I'm not sure I'm following. Could you show me an example, please?

**Comment by raunakab** [comment]
> > > hooray I agree 🕺
> > 
> > 
> > @andrewgazelka This is a large problem in our codebase. I'm going to start merging in small, piecemeal PRs to address this issue.
> 
> Do we want to prio this though?
> 
> _BigDaft_
> 
> ideally if we want to enforce something should also add to clippy lints https://rust-lang.github.io/rust-clippy/master/index.html

Probably not right now, but in the future.

File: src/daft-core/src/datatypes/matching.rs (hunk 1 above)
Comment (andrewgazelka) [reply src/daft-core/src/datatypes/matching.rs:49]: actually this does not matter. I misread; we are importing with

`use $crate::datatypes::DataType;`

**Review by andrewgazelka — APPROVED** [review]

**Pull request merged**